### PR TITLE
Make dungeons generated on demand rather than on start.

### PR DESCRIPTION
**Pull request by Predelnik**
Description is pretty obvious. I just want to note that it's tremendously speeds up start of program in debug and I do think it's important.

### CODE REVIEW

File: apps/freeablo/faworld/world.cpp [hunk 1]
```diff
@@ -151,7 +151,14 @@ namespace FAWorld
 
     GameLevel* World::getLevel(size_t level)
     {
-        return mLevels[level];
+       auto p = mLevels.find (level);
+       if (p == mLevels.end ())
+         return nullptr;
+       if (p->second == nullptr)
+         {
```
Comment (wheybags) [inline apps/freeablo/faworld/world.cpp]: dodgy indentation here, otherwise lgtm

**Comment by Predelnik** [comment]
I've fixed that. Btw is that ok if I do `git push -f` like here or is it better to add requested changes through new commits?

**Comment by wheybags** [comment]
That's fine, my policy is generally force-pushing prs is fine, but not
master

**Comment by wheybags** [comment]
bump :v

**Pull request merged**